models/user.go

```diff
@@ -0,0 +1,54 @@
+package models
+
+import (
+	"time"
+
+	"github.com/astaxie/beego/orm"
+	"github.com/astaxie/beego/validation"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type User struct {
+	Id             int
+	Email          string `orm:"unique"`
+	HashedPassword string
+
+	CreatedAt time.Time `orm:"auto_now_add;type(datetime)"`
+	UpdatedAt time.Time `orm:"auto_now;type(datetime)"`
+}
+
+func init() {
+	orm.RegisterModel(new(User))
+}
+
+func (u *User) TableName() string {
+	return "users"
+}
+
+func CreateUser(user *User, password string) error {
+	o := orm.NewOrm()
+
+	hashedPassword, _ := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	user.HashedPassword = string([]byte(hashedPassword))
+
+	valid := validate(user, password)
```

Comment:
Minor naming preference for `validation`:

```suggestion
	validation := validate(user, password)
```

`validation.HasErrors()` reads better to me 😁 Also I would expect `is_valid` as an attribute or predicate method but not as a noun.

Reply:
Fiexed in https://github.com/andyduong1920/go_marketer/pull/3/commits/b08a8906a9fe0341184f1d6398a17c7efb241397 @olivierobert